tracing/trace.go

```diff
@@ -49,16 +51,41 @@ type trace struct {
 	tags     map[string]string
 }
 
+func (t *trace) internalTracer() *Tracer {
+	if t == nil {
+		return nil
+	}
+	if tracer, ok := t.tracer.(*Tracer); ok && tracer != nil {
+		return tracer
+	}
+	return nil
+}
+
 func newTrace(tracer *Tracer, name string) *trace {
+	var (
+		otTracer        opentracing.Tracer
+		traceID, spanID string
+	)
 	if tracer == nil {
-		tracer = &globalTracer
+		otTracer = opentracing.GlobalTracer()
+	
+		// opentracing.Tracer is an interface that has only StartSpan, Inject, and Extract methods.
+		// This function expects a *tracing.Tracer, but the global opentracing.Tracer may not be one.
+		// That's why it's relying on the globalTracer: even if the opentracing.GlobalTracer() has been overridden,
+		// the IDs generated by baseplate will still conform to the specified configuration.
+		traceID = globalTracer.newTraceID()
```

Comment:
if we're not using globalTracer as a tracer, can we shield it somehow so that it only exposes the methods we want to be used on it, so it can't accidentally get missed again?